src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebApplicationTests.cs

```diff
@@ -591,35 +592,78 @@ public void WebApplicationBuilderHostBuilderSettingsThatAffectTheHostCannotBeMod
         }
 
         [Fact]
-        public void WebApplicationBuilderCanModifyWebRootAfterCreateBuilder()
+        public void WebApplicationBuilder_CanSetWebRootPaths()
         {
+            var contentRoot = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            var webRoot = "wwwroot2";
+            var webRootFullPath = Path.Combine(contentRoot, webRoot);
+            Directory.CreateDirectory(webRootFullPath);
+
             var builder = WebApplication.CreateBuilder();
 
-            var webRoot = Path.GetTempPath().ToString();
+            builder.WebHost.UseWebRoot(webRootFullPath);
+            Assert.Equal(webRootFullPath, builder.WebHost.GetSetting("webroot"));
 
-            builder.WebHost.UseSetting("WEBROOT", webRoot);
+            var app = builder.Build();
+            Assert.Equal(webRootFullPath, app.Environment.WebRootPath);
+        }
 
-            Assert.Equal(webRoot, builder.WebHost.GetSetting("webroot"));
+        [Fact]
+        public void WebApplicationBuilder_CanChangeSetWebRootPathsViaConfigureAppConfiguration()
+        {
+            var builder = WebApplication.CreateBuilder();
+            var webRootPath = "www";
+            var fullWebRootPath = Path.Combine(Directory.GetCurrentDirectory(), webRootPath);
```

Comment:
Can we use `WebApplicationOptions.ContentRootPath` and use a temp directory for this test? We should also be careful to delete all the test temp directories.